packages/babel-helper-annotate-as-pure/src/index.js

```diff
@@ -0,0 +1,19 @@
+import * as t from "babel-types";
+
+const PURE_ANNOTATION = "#__PURE__";
+
+const isPureAnnotated = node => {
+  const { leadingComments } = node;
+  if (leadingComments === undefined) {
+    return false;
+  }
+  return leadingComments.some(comment => /[@#]__PURE__/.test(comment.value));
```

Comment:
I wonder if there's a good way to prevent false positives here (so it doesn't match, say, "#__PURE___")

Reply:
actually this test is taken from the UglifyJS' source code, it tests it exactly like this, `foo #__PURE__ bar` are also allowed

Reply:
I think it would be convenient to have an PureAnnotation node in the Babel AST because It would probably be simpler to detect and manage. I assume babel/minify will also take advantage of it during minification.

Reply:
@xtuc I dont think thats actually possible, because ast -> code -> ast are not inverse operations (in mathematics sense) regarding comments

i.e. if we annotate each CallExpression with a leading comment in `fn()()()` we'll end up with nice AST - 1 comment for each node, when we generate code, we'll end up with
```js
/*foo*//*foo*//*foo*/fn()()()
```
And ofc if we go back to AST again we'll end up with 3 comments on the outermost CallExpression